### PR TITLE
Feature/allowlist behind reverse proxy

### PR DESCRIPTION
**Pull request by DCkQ6**
Added http request set-src directive to backend definition.

This feature is required to control access to specific paths while ingress is hidden behind reverse proxy. It is ment to use only with trusted headers, where any header modification is strictly controlled by reverse proxy and ingress itself is in separated - air gapped - network.

### CODE REVIEW

File: rootfs/etc/templates/haproxy/haproxy.tmpl [hunk 1]
```diff
@@ -512,6 +512,9 @@ backend {{ $backend.ID }}
 {{- range $e1 := short 10 $allow.Exception }}
     acl allow_exception_src{{ $i }} src{{ range $e := $e1 }} {{ $e }}{{ end }}
 {{- end }}
+{{- with $allow.SourceHeader }}
```
Comment (jcmoraisjr) [inline rootfs/etc/templates/haproxy/haproxy.tmpl:515]: Change to `if` instead just to follow the convention.

File: docs/content/en/docs/configuration/keys.md [hunk 2]
```diff
@@ -677,6 +678,7 @@ See also:
 | `allowlist-source-range` | `Path` |         | v0.12 |
 | `denylist-source-range`  | `Path` |         | v0.12 |
 | `whitelist-source-range` | `Path` |         |       |
+| `allowlist-source-header`| `Path` |         | v0.1x |
```
Comment (jcmoraisjr) [inline docs/content/en/docs/configuration/keys.md:681]: This can be merged to v0.13, so please change to v0.13.2 here.

**Review by jcmoraisjr — COMMENTED** [review]
Thanks for the contribution! Following some minor changes.

File: rootfs/etc/templates/haproxy/haproxy.tmpl (hunk 1 above)
Comment (DCkQ6) [reply rootfs/etc/templates/haproxy/haproxy.tmpl]: done :+1:

File: docs/content/en/docs/configuration/keys.md (hunk 2 above)
Comment (DCkQ6) [reply docs/content/en/docs/configuration/keys.md]: done :+1:

**Comment by jcmoraisjr** [comment]
LGTM, merging now, thanks!

**Pull request merged**